node/src/components/small_network/gossiped_address.rs

```diff
@@ -28,6 +50,32 @@ impl Display for GossipedAddress {
     }
 }
 
+impl ToBytes for GossipedAddress {
+    fn to_bytes(&self) -> Result<Vec<u8>, BytesreprError> {
+        let mut buf = allocate_buffer(self)?;
+        self.write_bytes(&mut buf)?;
+        Ok(buf)
+    }
+
+    fn serialized_length(&self) -> usize {
+        ToBytes::serialized_length(&self.0.to_string())
+    }
+
+    fn write_bytes(&self, writer: &mut Vec<u8>) -> Result<(), BytesreprError> {
+        self.0.to_string().write_bytes(writer)
```

Comment:
Hmm - this is a bit suboptimal.  The string representation is likely to be more expensive in terms of computation and space.

I think it would be better to implement bytesrepr traits for the following std types in the bytesrepr module of casper-types:
* `Ipv4Addr`
* `Ipv6Addr`
* `IpAddr`
* `SocketAddrV4`
* `SocketAddrV6`
* `SocketAddr`

We should use the [`octets()`](https://doc.rust-lang.org/std/net/struct.Ipv6Addr.html#method.octets) method on `Ipv4Addr` and `Ipv6Addr` to be efficient.

We should probably share a couple of const tags for the implementation of the `IpAddr` and `SocketAddr` enums - something like:
```rust
pub const IP_V4_TAG: u8 = 0;
pub const IP_V6_TAG: u8 = 1;
```